test/regExpTest.coffee

```diff
@@ -0,0 +1,71 @@
+expect = require('chai').expect
+
+describe 'regExp matches URI', ->
+  regExp = new RegExp("^(http[s]?:\\/\\/(www\\.)?|ftp:\\/\\/(www\\.)?|www\\.){1}([0-9A-Za-z-\\.@:%_\+~#=]+)+((\\.[a-zA-Z]{2,3})+)(/(.)*)?(\\?(.)*)?", "i")
```

Comment:
Previous Regex: 
```
Data {"url":"http://fakewebsite.com"}
```
Your Regex:
```
Data: {"url":"http://fakeweb"}
```

Your regex was working on other testing sites, but when I console logged the data it didn't contain the whole site URL